### PR TITLE
Fix(react): autosuggest prop not working correctly

### PR DESCRIPTION
**Pull request by anik-ghosh-au7**
- Issue Type - Bug

- Notion Link - https://www.notion.so/appbase/Searchbox-Fix-autosuggest-property-159e39a529004485b286845fc4ac3a22

- Description - Passing false to the autosuggest prop was still triggering the default query and showing the suggestions list.

- Fixes -
-> Don't trigger `defaultQuery` when `autosuggest` is `false`.
-> Don't fetch `recentSearches` when `autosuggest` is `false`.
-> Don't display suggestions when `autosuggest` is `false`.
-> When `autosuggest` is `false` then results should get updated on each key stroke.
-> Debounce should work as expected.

- Demo -

https://www.loom.com/share/d0524c13c37b43459f046608c2fa8e4d

### CODE REVIEW

**Review by bietkul — APPROVED** [review]

**Comment by bietkul** [comment]
@anik-ghosh-au7 This PR is not well tested. Just check your video at `1:37` it is showing the suggestion box with suggestions from URL params even when the `autosuggest` is `false`.
alt image

As I mentioned before in the Notion card this is not the way to test debounce. If you have debounce as 500ms then the next request should get triggered after 500ms irrespective of how fast the user types input.

**Review by bietkul — CHANGES_REQUESTED** [review]
@anik-ghosh-au7 Check the comments

**Comment by anik-ghosh-au7** [comment]
> @anik-ghosh-au7 Check the comments

@bietkul  made the changes also checked the denounce function  - https://www.loom.com/share/1b7572cca06e4888a29ad9b217e6b016

File: packages/react-searchbox/src/components/SearchBox.js [hunk 1]
```diff
@@ -64,10 +65,19 @@ class SearchBox extends React.Component {
   }
 
   componentDidMount() {
-    const { enableRecentSearches } = this.props;
-    if (enableRecentSearches) {
+    const { enableRecentSearches, autosuggest, id } = this.props;
+    if (enableRecentSearches && autosuggest) {
       this.componentInstance.getRecentSearches();
     }
+    if (window) {
```
Comment (bietkul) [inline packages/react-searchbox/src/components/SearchBox.js]: @anik-ghosh-au7 Why are you doing it here? URLParamsProvider would automatically handle it. Please refer the vue where you said it works.

**Comment by anik-ghosh-au7** [comment]
@bietkul demo - https://www.loom.com/share/781097e432ad4629baa2e91926f0eb60

**Review by bietkul — APPROVED** [review]

**Pull request merged**